### PR TITLE
ddl: add failpoints testcase

### PR DESCRIPTION
**Pull request by Benjamin2037**
### What problem does this PR solve?

Issue Number: ref #35983

Problem Summary:

### Check List

Tests 

- [X] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by tangenta — APPROVED** [review]

File: tests/realtikvtest/addindextest/common.go [hunk 1]
```diff
@@ -16,9 +16,11 @@ package addindextest
 
 import (
 	"context"
+	"github.com/pingcap/failpoint"
```
Comment (hawkingrei) [inline tests/realtikvtest/addindextest/common.go:19]: it needs to ```goimports```.

File: tests/realtikvtest/addindextest/BUILD.bazel [hunk 2]
```diff
@@ -24,6 +24,7 @@ go_test(
     srcs = [
         "add_index_test.go",
         "concurrent_ddl_test.go",
+        "failpoints_test.go"
```
Comment (hawkingrei) [inline tests/realtikvtest/addindextest/BUILD.bazel:27]: ```"failpoints_test.go",```

Comment (hawkingrei) [reply tests/realtikvtest/addindextest/BUILD.bazel:27]: It can updated automatically by ```gazelle```.

Comment (Benjamin2037) [reply tests/realtikvtest/addindextest/BUILD.bazel:27]: ok

**Review by hawkingrei — APPROVED** [review]

**Pull request merged**